### PR TITLE
Implement Button to create Desktop Shortcut

### LINKED ISSUE #447: Implement Button to create Desktop Shortcut
**Is your feature request related to a problem? Please describe.**
The implementation is already done on the backend, so just need to add some way to call it from the frontend.
Also we need an option under settings to create or not Desktop Shortcuts automatically, since now this is always true.

**Describe the solution you'd like**
- [ ] A button on the game page to create a Desktop Shortcut to launch the game
- [ ] A Setting to enable/disabled create Desktop shortcuts automatically when installing.

### PR DESCRIPTION
**Pull request by dragonDScript**
Same as #452

Use the following Checklist if you have changed something on the Backend or Frontend:

- [x] Created / Updated Tests
- [ ] Created / Updated documentation

Tested on a Clean Install with AppImage:
- [ ] Login
- [ ] Download a Game
- [ ] Import a Game
- [ ] Launch Default config
- [ ] Launch Personalized Config
- [ ] Launch wine
- [ ] Launch Proton
- [ ] Launch Custom Proton
- [ ] Launch Tools (Winetricks and Winecfg)
- [ ] Uninstall Game
- [ ] Move Game
- [ ] Languages (Test if no translation stopped working)

Fixes #447

### CODE REVIEW

**Comment by dragonDScript** [comment]
Not my fault: 
TypeScript error in /home/runner/work/HeroicGamesLauncher/HeroicGamesLauncher/src/screens/Settings/index.tsx(283,15):
Type '{ egsPath: string; setEgsPath: Dispatch<SetStateAction<string>>; egsLinkedPath: string; setEgsLinkedPath: Dispatch<SetStateAction<string>>; ... 15 more ...; discordRPC: boolean; }' is not assignable to type 'IntrinsicAttributes & Props'.
  Property 'toggleDiscordRPC' does not exist on type 'IntrinsicAttributes & Props'.  TS2322

    281 |               toggleAddDesktopShortcuts={toggleAddDesktopShortcuts.toggle}
    282 |               toggleAddGamesToStartMenu={toggleAddGamesToStartMenu.toggle}
  > 283 |               toggleDiscordRPC={toggleDiscordRPC}
        |               ^
    284 |               discordRPC={discordRPC}
    285 |             />
    286 |           )}

wtf

**Comment by Nocccer** [comment]
Checked main branch. Test passing locally. Create dist packages works. Can't reproduce your errors.

**Comment by dragonDScript** [comment]
Please use squash and merge for future PRs

**Comment by flavioislima** [comment]
> Please use squash and merge for future PRs

We  use it for merge into main.
i prefer it as well, but that was a long discussion some time ago. 
We can discuss this again since we have more devs now.

**Comment by dragonDScript** [comment]
> > Please use squash and merge for future PRs
> 
> We use it for merge into main.
> i prefer it as well, but that was a long discussion some time ago.
> We can discuss this again since we have more devs now.

Okay. Here are my arguments: squash and merge lets you revert a PR at any time.

**Comment by dragonDScript** [comment]
I fixed the conflicts.

**Comment by flavioislima** [comment]
> > > Please use squash and merge for future PRs
> > 
> > 
> > We use it for merge into main.
> > i prefer it as well, but that was a long discussion some time ago.
> > We can discuss this again since we have more devs now.
> 
> Okay. Here are my arguments: squash and merge lets you revert a PR at any time.

Yes, that's why we do it in the main branch. If you check the PRs merged there, there is a revert button. :)

But for PRs I don't know if its possible to revert a specific commit, only if you merge a PR on another PR.

**Comment by flavioislima** [comment]
> I fixed the conflicts.

i think there is still one left. To avoid conflicts with translations, is good not to change the locale files on a PR, since all of them are handled by weblate. 👍🏽

**Comment by dragonDScript** [comment]
I've fixed the conflicts and still there is one left

File: electron/legendary/games.ts [hunk 1]
```diff
@@ -462,35 +429,19 @@ Categories=Game;
...
   public stop() {
     // until the legendary bug gets fixed, kill legendary on mac
     // not a perfect solution but it's the only choice for now
+
     // @adityaruplaha: this is kinda arbitary and I don't understand it.
-    const pattern = process.platform === 'linux' ? this.appName : 'legendary'
+    const pattern = process.platform === 'darwin' ? 'legendary' : this.appName
     console.log('killing', pattern)
-
-    if (process.platform === 'win32'){
```
Comment (flavioislima) [inline electron/legendary/games.ts]: What happened here? Moved? Removed?

File: electron/main.ts [hunk 2]
```diff
@@ -617,15 +612,7 @@ ipcMain.handle(
 )
 
 ipcMain.handle('egsSync', async (event, args) => {
-  const egl_manifestPath = 'C:/ProgramData/Epic/EpicGamesLauncher/Data/Manifests'
```
Comment (flavioislima) [inline electron/main.ts]: Here as well. Looks like all windows fixes were removed.

File: electron/main.ts [hunk 3]
```diff
@@ -507,10 +506,6 @@ ipcMain.handle('openMessageBox', async (e, args) => {
   return {response}
 })
 
-ipcMain.handle('showErrorBox', async (e, args: [title: string, message: string]) => {
```
Comment (flavioislima) [inline electron/main.ts:510]: Moved or removed?

File: electron/legendary/games.ts [hunk 4]
```diff
@@ -355,50 +362,10 @@ Categories=Game;
       autoInstallDxvk
     } = await this.getSettings()
 
-    const DiscordRPC = makeClient('852942976564723722')
```
Comment (flavioislima) [inline electron/legendary/games.ts:358]: Is this elsewhere?

File: jest.config.js [hunk 5]
```diff
@@ -3,20 +3,48 @@ module.exports = {
     '**/*.{js,jsx,ts,tsx}',
     '!**/*.config.js'
   ],
+
   coverageDirectory: '<rootDir>/coverage',
+
   coveragePathIgnorePatterns: [
-    '<rootDir>/node_modules',
+    '<rootDir>/node_modules>',
```
Comment (flavioislima) [inline jest.config.js:10]: I this line have an error. The > at the end I guess.

**Review by flavioislima — CHANGES_REQUESTED** [review]

**Review by flavioislima — APPROVED** [review]
Looks good now 👍

**Pull request merged**